### PR TITLE
Redesign tracing page toolbar

### PR DESCRIPTION
**Pull request by jotak**
- Toolbar entirely grouped into a single dropdown, like in graph
- Some removed fields: trace intervals, status code
- Some new fields: percentiles, show spans average
- New component TracesDisplayOptions
- Some styles from GraphSettings are moved to a common place and made
  accessible to TracesDisplayOptions
- removed now useless function, like buildTags, which btw had a lib
  dependency that could be removed as well

Fixes https://github.com/kiali/kiali/issues/3604

Capture d’écran de 2021-01-27 15-39-22

### CODE REVIEW

File: src/components/JaegerIntegration/TracesComponent.tsx [hunk 1]
```diff
@@ -153,130 +99,61 @@ class TracesComponent extends React.Component<TracesProps, TracesState> {
...
+    const tags = this.getTags();
     if (tags) {
       url += `&tags=${tags}`;
     }
     return url;
   };
 
-  private handleStatusCode = (value: string) => {
+  private onQuerySettingsChanged = (settings: QuerySettings) => {
+    console.log(settings);
```
Comment (jotak) [inline src/components/JaegerIntegration/TracesComponent.tsx:145]: oops; // FIXME / REMOVE

File: src/components/JaegerIntegration/TracesComponent.tsx [hunk 2]
```diff
@@ -153,130 +99,61 @@ class TracesComponent extends React.Component<TracesProps, TracesState> {
...
-
-  private filterTraces = (): JaegerTrace[] => {
-    if (!this.state.selectedIntervalDuration) {
-      return this.state.traces;
-    }
-    const min = this.state.selectedIntervalDuration.min * this.state.selectedIntervalDuration.scale;
-    const max = this.state.selectedIntervalDuration.max * this.state.selectedIntervalDuration.scale;
-    return this.state.traces.filter(trace => trace.duration >= min && trace.duration <= max);
+  private onDisplaySettingsChanged = (settings: DisplaySettings) => {
+    console.log(settings);
```
Comment (jotak) [inline src/components/JaegerIntegration/TracesComponent.tsx:151]: oops; // FIXME / REMOVE

**Comment by jotak** [comment]
Update: new menu:
Capture d’écran de 2021-01-28 16-01-27

- Percentile values are displayed for reference
- Removed Time axis options as they're aren't very relevant anymore

**Comment by jotak** [comment]
PS: don't hesitate to comment on texts / labels also

**Review by hhovsepy — COMMENTED** [review]
Verified the functionality.
Just line up of checkbox and values will look better.

**Comment by jotak** [comment]
Thanks @hhovsepy 

> Just line up of checkbox and values will look better.

Not sure if you mean to add a separator line between option groups, in case yes perhaps that would be better, however I've copied the design from the graph display dropdown, where there isn't anything like that

**Comment by hhovsepy** [comment]
I mean the label text of checkbox or radiobutton are a bit higher:
Screenshot from 2021-01-29 10-37-00

**Comment by jotak** [comment]
@jshaughn can you have a quick look at my last commit https://github.com/kiali/kiali-ui/pull/2066/commits/7a1b49ff46d549bbf83be5f859cb71aca0bbddd7 ?
Since I'm reusing here styles from the graph, this is affecting graph code (but should have no visible effect). I'm moving a couple of CSS style to typestyle.

**Comment by jotak** [comment]
> I mean the label text of checkbox or radiobutton are a bit higher:

I see, it should be fixed now

**Review by hhovsepy — APPROVED** [review]
Verified now:
Screenshot from 2021-01-29 12-31-00

**Review by jshaughn — CHANGES_REQUESTED** [review]
> @jshaughn can you have a quick look at my last commit 7a1b49f ?

@jotak,  I don't think the CSS change you pointed out in that commit is an issue, the alignment of the dropdown margin for the checkbox or radio elements looks unchanged.  But there does seem to be a regression in the graph Display dropdown, I think due to a margin change around the info icons:

image

Before:
image


After:

image

**Comment by jotak** [comment]
@jshaughn good catch! fixed

**Comment by lucasponce** [comment]
A first comment, I see this new toaster when clicking in a trace from the graph:
image

It's odd as I guess I didn't click anything special so I don't see need for this warning.

**Comment by lucasponce** [comment]
Minor, I think the help icons could follow same pattern used in Graph Display:
image

And to be placed on value and at right:
image

**Comment by lucasponce** [comment]
I see a problem in the graph, the versioned app graph doesn't work well on my local env.
I'm going to test master and check if the problem is still there.

File: package.json [hunk 3]
```diff
@@ -75,7 +75,6 @@
     "js-yaml": "3.13.1",
     "json-beautify": "1.0.1",
     "lodash": "4.17.20",
-    "logfmt": "1.2.1",
```
Comment (lucasponce) [inline package.json:78]: unused ?

**Comment by jotak** [comment]
> Minor, I think the help icons could follow same pattern used in Graph Display:
> And to be placed on value and at right:

It can definitely be placed to the right, but it's meant to give information on the group of options, not each single value.

**Comment by jotak** [comment]
> A first comment, I see this new toaster when clicking in a trace from the graph:

Good catch, I didn't notice, it's because it's fetching inbound+outbound percentiles ... but only inbound queries is allowed on services (so you wouldn't see this warning on workloads or apps traces).
Anyway, it's now fixed for services.

File: src/components/JaegerIntegration/TracesComponent.tsx [hunk 4]
```diff
@@ -379,8 +262,9 @@ class TracesComponent extends React.Component<TracesProps, TracesState> {
                     </Toolbar>
                   </RenderHeader>
                   <JaegerScatter
-                    fixedTime={!this.state.adjustTime}
-                    traces={this.filterTraces()}
+                    fixedTime={false}
```
Comment (lucasponce) [inline src/components/JaegerIntegration/TracesComponent.tsx:265]: I guess only fixedTime was used for Jaeger I think.
Perhaps then that parameter could be removed, but it's fine.

**Review by lucasponce — APPROVED** [review]
Tested locally and code looks ok to me.
I didn't see any blockers.
I'll wait that @jshaughn is ok with it.

**Review by aljesusg — APPROVED** [review]
LGFM, sorry I had it in pending :( tested it !!!

**Review by jshaughn — APPROVED** [review]
LGTM!

File: src/components/JaegerIntegration/TracesComponent.tsx (hunk 4 above)
Comment (jotak) [reply src/components/JaegerIntegration/TracesComponent.tsx]: +1, I've removed it

**Comment by jotak** [comment]
Thanks all

**Pull request merged**